### PR TITLE
Handle Fixidity in smart contracts

### PR DESCRIPTION
**Pull request by m-chrzan**
### Description

This is a sister PR to a [monorepo PR](https://github.com/celo-org/celo-monorepo/pull/337) that replaces our exact fractions library with Fixidity - a fixed point arithmetic library.

Changes in celo-blockchain:

* updated intrinsic gas costs
* to reduce impact on this repo for now, didn't implement handling of fixed point numbers from smart contracts. The one place where this matters is handling of the infrastructure fraction. The contract keeps a fixed point `infrastructureFraction` and exposes an `infrastructureFraction_` function that returns a numerator and denominator.

### Tested

Monorepo e2e tests pass.

### Related issues

- Part of https://github.com/celo-org/celo-monorepo/issues/166

### Backwards compatibility

Incompatible with previous networks since it increases gas costs.

### CODE REVIEW

**Comment by ashishb** [comment]
Tests are failing - https://circleci.com/gh/celo-org/celo-blockchain/6454

**Comment by m-chrzan** [comment]
> Tests are failing

Hopefully this is just due to Circle using the not updated celo-blockchain version. Updated Circle config to test against my branch, thanks for flagging.

File: params/protocol_params.go [hunk 1]
```diff
@@ -129,10 +129,10 @@ const (
 	// TODO(asa): Make these operations less expensive by charging only what is used.
 	// The problem is we don't know how much to refund until the refund is complete.
 	// If these values are changed, "setDefaults" will need updating.
-	MaxGasForDebitFromTransactions      uint64 = 50 * 1000
+	MaxGasForDebitFromTransactions      uint64 = 60 * 1000
```
Comment (asaj) [inline params/protocol_params.go]: Why are such large increases necessary?

File: .circleci/config.yml [hunk 2]
```diff
@@ -80,7 +80,7 @@ jobs:
           command: |
             set -euo pipefail
             export CELO_MONOREPO_DIR="$HOME/geth/celo-monorepo"
-            git clone --depth 1 https://github.com/celo-org/celo-monorepo.git ${CELO_MONOREPO_DIR} -b master
+            git clone --depth 1 https://github.com/celo-org/celo-monorepo.git ${CELO_MONOREPO_DIR} -b m-chrzan/fixidity
```
Comment (asaj) [inline .circleci/config.yml]: You only need to repoint the e2e tests for the *first* PR that you merge, and you can leave the other pointing to master.

File: params/protocol_params.go [hunk 3]
```diff
@@ -129,10 +129,10 @@ const (
...
 	// The problem is we don't know how much to refund until the refund is complete.
 	// If these values are changed, "setDefaults" will need updating.
-	MaxGasForDebitFromTransactions      uint64 = 50 * 1000
-	ExpectedGasForDebitFromTransactions uint64 = 35 * 1000
-	MaxGasForCreditToTransactions       uint64 = 30 * 1000
-	MaxGasToReadErc20Balance            uint64 = 10 * 1000
+	MaxGasForDebitFromTransactions      uint64 = 48 * 1000
+	ExpectedGasForDebitFromTransactions uint64 = 32 * 1000
+	MaxGasForCreditToTransactions       uint64 = 38 * 1000
+	MaxGasToReadErc20Balance            uint64 = 25 * 1000
```
Comment (asaj) [inline params/protocol_params.go]: How did you calculate these numbers?

Comment (asaj) [inline params/protocol_params.go]: Why did `MaxGasToReadErc20Balance` go up so much?

File: .circleci/config.yml (hunk 2 above)
Comment (asaj) [reply .circleci/config.yml]: Still suggest this

**Pull request merged**